packages/client/src/client.ts

```diff
@@ -87,18 +83,21 @@ export class SkipClient {
     getRpcEndpointForChain?: (chainID: string) => Promise<string>;
     getRestEndpointForChain?: (chainID: string) => Promise<string>;
   };
-
-  protected getCosmosSigner?: clientTypes.SignerGetters["getCosmosSigner"];
-  protected getEVMSigner?: clientTypes.SignerGetters["getEVMSigner"];
-  protected getSVMSigner?: clientTypes.SignerGetters["getSVMSigner"];
-  protected chainIDsToAffiliates?: clientTypes.SkipClientOptions["chainIDsToAffiliates"];
-  protected cumulativeAffiliateFeeBPS?: string = "0";
-  protected cacheDurationMs?: number;
-
-  private cache: CustomCache;
-  private cachingMiddleware: ReturnType<typeof createCachingMiddleware>;
+  getCosmosSigner?: clientTypes.SignerGetters["getCosmosSigner"];
+  getEVMSigner?: clientTypes.SignerGetters["getEVMSigner"];
+  getSVMSigner?: clientTypes.SignerGetters["getSVMSigner"];
+  chainIDsToAffiliates?: clientTypes.SkipClientOptions["chainIDsToAffiliates"];
+  cumulativeAffiliateFeeBPS?: string = "0";
+
+  private clientOptions: clientTypes.SkipClientOptions;
+  private skipChains?: types.Chain[];
+  private skipAssets?: Record<string, types.Asset[]>;
+  private skipBalances?: types.BalanceResponse;
```

Comment:
this var will have an edge case if the user change the address. we will need an address validation in executeRoute to check the address in skipBalances.chains response is same, if not refetch

Reply:
for example: 
wallet a: have osmo on osmosis
wallet b: don't have osmo on osmosis

executeRoute using wallet b fee asset is osmo on osmosis:
it will assume the asset is available

Reply:
I think once you change your address and you fetch the new balance it will update the cache again, so unless they manage to call executeRoute before fetching the balance of wallet b I think it won't cause issues?

Reply:
because balances is merged like this https://github.com/skip-mev/skip-go/pull/875/files#diff-1d25041624e23bb1be334f0c7033cf4fadf2801dae21687f77f5511cd5935b92R277-R280

Reply:
it will only check the key which is the chain id

Reply:
i think the new one will overwrite the old one

Reply:
its easier to not merge them then, and always just replace the old skipBalances

Reply:
![Uploading image.png…]()

Reply:
![image](https://github.com/user-attachments/assets/6e5ce90a-ad7a-4c8d-9b89-b3a0e65b7425)

Reply:
yes

Reply:
updated!